test/connection.js

```diff
@@ -0,0 +1,51 @@
+/* eslint-env node, mocha */
+require('./init.js');
+var assert = require('assert');
+var DataSource = require('loopback-datasource-juggler').DataSource;
+
+var config;
+
+before(function() {
+  config = global.getConfig();
+  console.log("config", config);
+});
+
+describe('testConnection', function() {
+  it('should pass with valid settings', function(done) {
+    var db = new DataSource(require('../'), config);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
+      done(err);
+    });
+  });
+
+  it('should pass when valid URL overrides invalid settings', function(done) {
+    var url = generateURL(config);
+    var dbConfig = {
+      url: url,
+      host: 'invalid-hostname',
+      port: 80,
+      database: 'invalid-database',
+      username: 'invalid-username',
+      password: 'invalid-password',
+    };
+
+    var db = new DataSource(require('../'), dbConfig);
+    db.ping(function(err) {
+      assert(!err, 'Should connect without err.');
+      done(err);
+    });
+  });
+});
+
+function generateURL(config) {
+  var url =
+    'mssql://' + config.user +
+    ':' + config.password +
+    '@' + config.host +
+    ':' + config.port +
+    '/' + config.database +
```

Comment:
This assumes that the config values don't contain any special characters. A safer way is to call `url.format`, see https://nodejs.org/docs/latest/api/url.html#url_url_format_urlobj